model/progress.go

```diff
@@ -0,0 +1,7 @@
+package model
+
+type StreamProgress struct {
+	Progress float64
+	StreamID uint `gorm:"primaryKey"`
+	UserID uint `gorm:"primaryKey"`
```

Comment:
It's not entirely clear why this needs to be a primary key (I see why it does but it could use a helpful comment)

Reply:
Yes, I also got confused and thought using foreign keys was possible when coming back for refactoring. I will add a comment.